applications/tari_base_node/src/builder.rs

```diff
@@ -214,7 +215,7 @@ pub fn configure_and_initialize_node(
     config: &GlobalConfig,
     id: NodeIdentity,
     rt: &mut Runtime,
-) -> Result<(CommsNode, NodeType, MinerType, Wallet), String>
+) -> Result<(CommsNode, NodeType, MinerType, Interfaces), String>
```

Comment:
Since you've started down this road, I would throw more things in `Interfaces`, incl the comms node at least. 

Usually this kind of object is called the execution context, so naming it `BaseNodeContext` rather than `Interfaces`

Reply:
I've renamed it, but I will add the interface handles when I require them. I know those 3 I require currently.